test/tests/functional/pbs_suspend_resume_accounting.py

```diff
@@ -0,0 +1,356 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2020 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.functional import *
+
+
+class TestSuspendResumeAccounting(TestFunctional):
+    """
+    Testsuite for verifying accounting of
+    suspend/resume events of job
+    """
+
+    script = ['#!/bin/bash\nfor ((c=1; c <= 1000000000; c++));']
+    script += ['do']
+    script += ['sleep 1']
+    script += ['done']
+
+    def test_suspend_resume_job_signal(self):
+        """
+        Test case to verify accounting suspend
+        and resume records when the events are
+        triggered by client for one job.
+        """
+        j = Job(TEST_USER)
+        j.create_script(self.script)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+
+        self.server.sigjob(jobid=jid, signal="suspend")
+
+        record = 'z;%s;requestor=%s@%s .*resources_used.' % \
+                 (jid, self.server.current_user, self.server.hostname)
+        self.server.accounting_match(msg=record, id=jid, regexp=True)
+
+        self.server.sigjob(jobid=jid, signal="resume")
+        record = 'r;%s;requestor=Scheduler@%s' % \
```

Comment:
Why is the requestor "Scheduler" if the user is resuming it via qsig ?

Reply:
When a user does a qsig -s resume, it moves the job from suspended substate 43 (user suspend) to suspended substate 45 (scheduler suspended), but does not resume the job.  It is the scheduler that does the resume when resources are available.

Reply:
Thanks @bhroam. I guess the requestor is scheduler here because I am logging after the resume was successful by the scheduler i.e. post_signal processing. Should we also record user requested resume?

Reply:
if the requester is always going to be Scheduler for resume record then there isn't really a lot of value logging it.

Reply:
In case of `admin-resume`, the job is resumed directly, without waiting for the scheduler and it could be a different user. So, maybe I think we should keep it.

The user requested resume is kind of "attempted-resume" which may or may not succeed always. What if we have an additional record to say that there was a user requested resume?

Reply:
nah, that can be confusing. Since the request to resume was initiated by the user, I think the right thing to do is to print the actual user's name as the requester, even if it happens via the Scheduler. So, if possible, I think we should print the user's name as the requester.

What do you think @bhroam  ?